SWIG/creditdefaultswap.i

```diff
@@ -58,12 +80,20 @@ class CreditDefaultSwapPtr : public boost::shared_ptr<Instrument> {
                              BusinessDayConvention paymentConvention,
                              const DayCounter& dayCounter,
                              bool settlesAccrual = true,
-                             bool paysAtDefaultTime = true) {
+                             bool paysAtDefaultTime = true,
+			     const Date& protectionStart = Date(),
+			     const Date& upfrontDate = Date(),
+			     const DayCounter& lastPeriodDayCounter = DayCounter(),
+                             const bool rebatesAccrual = true,
+                             const ClaimPtr& claim = ClaimPtr()) {
+            boost::shared_ptr<Claim> cdsClaim = boost::dynamic_pointer_cast<Claim>(claim);
```

Comment:
I don't think the cast is needed; `claim` is already a `shared_ptr<Claim>`.  However, there's no way for the user to create one.  If we want to actually pass the argument instead of using the default one, we should export the derived claim classes and their constructors.

Reply:
Yeah I see what you mean. The only reason I included any of the claim language was because it appears as an optional input before lastPeriodDayCounter and rebatesAccrual in the C++ CreditDefaultSwap constructor. So I just put in a default value in there as a place filler since I didn't actually need that functionality to replicate the code in the C++ test-suite.

Reply:
I see.  I'd prefer the parameters to keep the same order as in C++, though.  But I can fix them when I merge.